From f9d5f744bc1a38ebb86683ecaa9775ea0e6c769f Mon Sep 17 00:00:00 2001
From: Ruslan Bay <ruslanbey@proton.me>
Date: Thu, 26 Feb 2026 17:21:22 +0100
Subject: [PATCH 26/54] media: intel/ipu4: Update try_to_del_timer_sync() to
 timer_delete_sync_try()

Replace try_to_del_timer_sync() with timer_delete_sync_try(), which is the new
kernel API for attempting to delete timers synchronously.
---
 drivers/media/pci/intel/ipu4/ipu4-psys.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/pci/intel/ipu4/ipu4-psys.c b/drivers/media/pci/intel/ipu4/ipu4-psys.c
index 74309962b152..06312ac5bcf9 100644
--- a/drivers/media/pci/intel/ipu4/ipu4-psys.c
+++ b/drivers/media/pci/intel/ipu4/ipu4-psys.c
@@ -374,7 +374,7 @@ void ipu_psys_kcmd_complete(struct ipu_psys *psys,
 
 	switch (kcmd->state) {
 	case KCMD_STATE_RUNNING:
-		if (try_to_del_timer_sync(&kcmd->watchdog) < 0) {
+		if (timer_delete_sync_try(&kcmd->watchdog) < 0) {
 			dev_err(&psys->adev->dev,
 				"could not cancel kcmd timer\n");
 			return;
-- 
2.51.0

